libdnf/rpm/solv/goal_private.hpp

```diff
@@ -173,10 +173,22 @@ inline void GoalPrivate::add_remove(const SolvMap & solv_map, bool clean_deps) {
     }
 }
 
-inline void GoalPrivate::add_upgrade(IdQueue & queue) {
+inline void GoalPrivate::add_upgrade(IdQueue & queue, bool best, bool clean_deps) {
     // TODO dnf_sack_make_provides_ready(sack); When provides recomputed job musy be empty
     Id what = pool_queuetowhatprovides(pool, &queue.get_queue());
-    staging.push_back(SOLVER_UPDATE | SOLVER_SOLVABLE_ONE_OF | SOLVER_SETARCH | SOLVER_SETEVR, what);
+    staging.push_back(
+        SOLVER_UPDATE | SOLVER_SOLVABLE_ONE_OF | SOLVER_SETARCH | SOLVER_SETEVR | (best ? 0 : SOLVER_FORCEBEST) |
```

Comment:
Really confusing. At first glance, it seems the opposite. `best ? SOLVER_FORCEBEST : 0`. I see it on 3 places. I'm confused or it is bug?